src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,142 @@
+import * as classNames from 'classnames';
+import * as PropTypes from 'prop-types';
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+import Button, { ButtonProps } from '../Button/Button';
+import * as Rivet from '../Rivet';
+import ModalEvent from './ModalEvent';
+
+
+export interface ModalProps {
+    /**
+     * Determines whether the modal is shown or not
+     */
+    isOpen?: boolean;
+    /**
+     * Optional function to call to raise when the alert is dismissed. If undefined, the modal will act as a dialog (i.e., no close button, 
+     * will not close on outside click or escape key)
+     */
+    onDismiss?: Rivet.Action;
+    /**
+     * The content of the modal's header
+     */
+    title: React.ReactNode | string;
+    /**
+     * Button or buttons which will appear in the footer of the modal
+     */
+    controls: React.ReactElement<ButtonProps> | Array<React.ReactElement<ButtonProps>>;
+}
+
+
+class Modal extends React.PureComponent<ModalProps & React.HTMLAttributes<HTMLButtonElement>> {
+    
+    public static displayName = 'Modal';
```

Comment:
Are you sure `displayName` is required?  I thought it was only needed for stateless components.